packages/twenty-server/src/workspace/messaging/services/gmail-partial-sync.service.ts

```diff
@@ -207,12 +244,33 @@ export class GmailPartialSyncService {
         maxResults,
       });
 
-      return { history: history.data };
+      let nextPageToken = history.data.nextPageToken;
+
+      const historyId = history.data.historyId;
+
+      if (history.data.history) fullHistory.push(...history.data.history);
+
+      while (nextPageToken) {
+        const nextHistory = await gmailClient.users.history.list({
+          userId: 'me',
+          startHistoryId: lastSyncHistoryId,
+          historyTypes: ['messageAdded', 'messageDeleted'],
+          maxResults,
+          pageToken: nextPageToken,
+        });
+
+        nextPageToken = nextHistory.data.nextPageToken;
+
+        if (nextHistory.data.history)
```

Comment:
```suggestion
        if (nextHistory.data.history) {
            fullHistory.push(...nextHistory.data.history);
         }
```
For better readability